pyomo/contrib/appsi/solvers/gurobi.py

```diff
@@ -274,8 +276,10 @@ def release_license(self):
                 gurobipy.disposeDefaultEnv()
 
     def __del__(self):
-        if not python_is_shutting_down():
-            self.release_license()
+        self._num_instances -= 1
+        if self._num_instances == 0:
+            if not python_is_shutting_down():
```

Comment:
You almost certainly need to check python_is_shutting_down before _num_instances (an when python is shutting down, the _num_instances attribute may have already been released)

Reply:
Great point. Thanks!